samcli/lib/init/__init__.py

```diff
@@ -131,6 +134,8 @@ def generate_project(
 
     _enable_application_insights(application_insights, output_dir, name)
 
+    _create_default_samconfig(package_type, output_dir, name)
```

Comment:
TODO: we need to address these kind of post-processing steps with hooks in the future.